ledger/from-archive/src/main.rs

```diff
@@ -41,54 +51,242 @@ fn main() {
         )
         .expect("Could not append origin block to ledger");
 
-    // Sync all blocks
-    let mut block_index = 1;
+    let mut start = 1;
+    let mut reached_end = false;
     loop {
+        let end = start + config.block_chunk_size;
```

Comment:
Shouldn't this be `min(start + config.block_chunk_size, config.num_blocks.unwrap_or(u64::maxint))`?